plugins/database/postgresql/postgresql_test.go

```diff
@@ -990,3 +992,151 @@ func TestNewUser_CustomUsername(t *testing.T) {
 		})
 	}
 }
+
+// This is a long-running integration test which tests the functionality of Postgres's multi-host
+// connection strings. It uses two Postgres containers preconfigured with Replication Manager
+// provided by Bitnami. This test currently does not run in CI and must be run manually. This is
+// due to the test length, as it requires multiple sleep calls to ensure cluster setup and
+// primary node failover occurs before the test steps continue.
+//
+// To run the test, set the environment variable POSTGRES_MULTIHOST_NET to the value of
+// a docker network you've preconfigured, e.g.
+// `docker network create -d bridge postgres-repmgr`
+// `export POSTGRES_MULTIHOST_NET=postgres-repmgr`
+func TestPostgreSQL_Repmgr(t *testing.T) {
+	_, exists := os.LookupEnv("POSTGRES_MULTIHOST_NET")
+	if !exists {
+		t.Skipf("POSTGRES_MULTIHOST_NET not set, skipping test")
+	}
+
+	// Create 2 postgres-repmgr containers
+	db0, runner0, url0, container0 := testPostgreSQL_Repmgr_Container(t, "psql-repl-0")
+	_, _, url1, _ := testPostgreSQL_Repmgr_Container(t, "psql-repl-1")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cancel()
+	time.Sleep(20 * time.Second)
```

Comment:
nit: I assume this is waiting for replication manager? Is it possible to query the containers instead of sleeping? I am wondering if sleeping here could make this test flaky?

When running postgres locally, usually I do something like

```bash
docker run --rm --net host postgres \
    psql postgresql://dbuser:dbpass@localhost:5432/postgres?sslmode=disable \
    -c "\x on;" \
    -c "select 1"
```

Not sure if that is possible with this docker API.

Reply:
I agree this would be ideal, but it looks like it would be a lot of changes to get arbitrary queries like this working. Not sure if it's worth going down that rabbit hole for this one test.

Reply:
I actually revisited this and got pretty close. I can run arbitrary queries which is good because there's an event table which could theoretically be checked for knowing when things are ready or not. But parsing the data is another challenge. Instead I tightened the sleep timings more to what should still be plenty of time for the cluster processing, yet still let the test finish in a reasonable amount of time.